### ISSUE TITLE
Error: OllmGenerateResponse variant discriminator field cannot be omitted

### ISSUE THREAD

**Issue by Luyten-Orion** [guzba/sunny#10]
Hey there, I'm currently writing an Ollama API wrapper and I decided to use `sunny` since it provided the features I needed, but when I added the `createdAt` field, Sunny complained about the omission of the discriminator field. Removing this field fixes the error but prevents me from being able to store the parsed timestamp, other than this, `sunny` is great and exactly what I needed!

```nim
import sunny, times

type
  OllmGenerateResponse* = object
    model* {.json: ",required".}: string
    createdAtStr {.json: "created_at,required".}: string
    createdAt {.json: "-".}: DateTime
    response* {.json: ",required".}: string
    case done*: bool
      of true:
        totalDuration* {.json: "total_duration,required".}: int64
        loadDuration* {.json: "load_duration,required".}: int64
        promptEvalCount* {.json: "prompt_eval_count,required".}: int64
        promptEvalDuration* {.json: "prompt_eval_duration,required".}: int64
        evalCount* {.json: "eval_count,required".}: int64
        evalDuration* {.json: "eval_duration,required".}: int64
        context* {.json: ",required".}: seq[int]
      else:
        discard

echo OllmGenerateResponse.fromJson("""{"model":"stuehieyr/nandine:latest","created_at":"2024-06-25T22:38:48.566852763Z","response":"Hello! How can I assist you today?\n","done":true,"total_duration":0,"load_duration":0,"prompt_eval_count":0,"prompt_eval_duration":0,"eval_count":0,"eval_duration":0,"context":[]}""")
```

**Comment by Luyten-Orion** [comment]
I forgot to include the exact error, the error is
```nim
/home/chronos/.nimble/pkgs2/sunny-0.1.8-4692a8890fa287ff182f347530f455bfbf8b7669/sunny.nim(1127, 9) template/generic instantiation of `fromJson` from here
/home/chronos/.nimble/pkgs2/sunny-0.1.8-4692a8890fa287ff182f347530f455bfbf8b7669/sunny.nim(1037, 22) Error: OllmGenerateResponse variant discriminator field cannot be omitted
```

**Comment by guzba** [comment]
Thanks for the heads up, small issue with a quick easy fix.

Fixed and tagged in 0.1.9 https://github.com/guzba/sunny/releases/tag/0.1.9

**Comment by Luyten-Orion** [comment]
Thanks for the quick reply and fix, heavily appreciated!
